src/trayitem.c

```diff
@@ -182,15 +194,19 @@ GtkWidget *tray_item_new( SniItem *sni, GtkWidget *tray )
   self = GTK_WIDGET(g_object_new(tray_item_get_type(), NULL));
   priv = tray_item_get_instance_private(TRAY_ITEM(self));
 
+  priv->box = gtk_box_new(GTK_ORIENTATION_HORIZONTAL,0);
```

Comment:
I suggest using gtk_grid_new() instead of a button. This will be more future proof (gtk_box is deprecated in gtk4) and will allow to control relative positioning of the label vis a vis the icon using -GtkWidget-direction property. This is the approach we already take with taskbar items (see function taskbar_item_new).
I also wonder if we should wrap the grid into a button, since we're tweaking the layout and associated CSS theming for tray items with this change. A button would make drag and drop implementation cleaner, but would require a CSS update for all existing configs to maintain the visuals.

Reply:
> I suggest using gtk_grid_new() instead of a --button--box

Will change.

> I also wonder if we should wrap the grid into a button

I am indifferent on this one.

Reply:
> > I also wonder if we should wrap the grid into a button
> 
> I am indifferent on this one.

Let's wrap it in a button, this will make flow items more consistent will result in a clear drag and drop implementation.

Reply:
Changed from box into a grid, for the button wrap it might be the best if you change that later,
I started implementing it but failed to wire up the signals correctly and then detected that there is an sfwbar internal implementation of a button as well.

Reply:
That's fine. I'm happy to code up the box wrapper. I had to mangle the signals quite a bit to make non-button widgets play nicely with drag and drop, so it's rather messy. I'll also replace the gtk_container_add's with gtk_grid_attach_next_to's (since that will require a button widget to pull the direction property from).